addons/binding/org.openhab.binding.icloud/README.md

```diff
@@ -0,0 +1,136 @@
+# iCloud Binding
+
+This binding uses the Apple [iCloud](https://www.apple.com/icloud/) service for providing device based information such as battery details or location.
+An Apple account is required. Two factor authentication is supported.
+
+Please note: Application specific passwords are not supported.
+
+The binding uses Google's [Geocode API](https://developers.google.com/maps/documentation/geocoding/) to derive a postal address from geographic coordinates.
+
+## Supported Things
+
+The following devices are known to work with this binding:
+
+* iPhone 6s
+* iPhone 5c
+* iPhone 7
+* iWatch 2
+* iPad Air 2/2017/Pro
+* MacBook Pro
+
+Other devices connected to your iCloud account should work as well.
+Please provide feedback if you have tested another device type.
+
+## Discovery
+
+For each iCloud account to be included an iCloud binding thing needs to be configured with your ID and password.
+The devices registered to this account will then be automatically discovered.
+
+## Binding Configuration
+
+The binding has no configuration options, all configuration is done at Thing level.
+
+## Thing Configuration
+
+The binding provides two thing types.
+
+### iCloud Account (Bridge) Thing
+
+The following table lists the configuration parameters:
+
+| Parameter            | Description                                         |                             |
+|----------------------|-----------------------------------------------------|-----------------------------|
+| appleId              |                                                     | mandatory                   |
+| password             |                                                     | mandatory                   |
+| refreshTimeInMinutes |                                                     | optional, 5 minutes default |
+| googleAPIKey         | Google API key to be used for address lookup calls. | optional                    |
+
+Without a "GoogleAPIKey" the number of daily request are limited (see also [API useage limits](https://developers.google.com/maps/documentation/geocoding/usage-limitsv)) and the bindings address lookup functionality will stop working.
+It is recommended to provide this value.
+
+A device is identified by a hash value calculated from the device id provided by apple.
+If a device is removed from the account the respective openHAB thing will go OFFLINE.
+
+## Channels
+
+### Account
+
+| Channel ID      | Type   | Description                                                  |
+|-----------------|--------|--------------------------------------------------------------|
+| numberOfDevices | Number | Number of registered devices with this iCloud account.       |
```

Comment:
May I ask: What is this one actually for? Consider removing it OR instead of providing the number of devices provide the deviceIds of devices.

Reply:
Number of device channel removed.

Reply:
Removed channel "numberOfDevices".